src/main/java/org/apache/accumulo/testing/continuous/ManySplits.java

```diff
@@ -0,0 +1,254 @@
+                splitThresholdStr);
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        });
+
+        log.info("Waiting for all tablets to have a sum file size <= {}", splitThreshold);
+
+        // wait for all tablets to reach the expected sum file size
+        tableNames.stream().parallel().forEach(tableName -> {
+          int elapsedMillis = 0;
+          long sleepMillis = SECONDS.toMillis(1);
+          try {
+            // wait for each tablet to reach the expected sum file size
+            while (true) {
+              Collection<Long> tabletFileSizes = getTabletFileSizes(client, tableName).values();
+              // filter out the tablets that are already the expected size
+              Set<Long> offendingTabletSizes =
+                  tabletFileSizes.stream().filter(tabletFileSize -> tabletFileSize > splitThreshold)
+                      .collect(Collectors.toSet());
+              // if all tablets are good, move on
+              if (offendingTabletSizes.isEmpty()) {
+                break;
+              }
+
+              elapsedMillis += sleepMillis;
+              // log every 3 seconds
+              if (elapsedMillis % SECONDS.toMillis(3) == 0) {
+                double averageFileSize =
+                    offendingTabletSizes.stream().mapToLong(l -> l).average().orElse(0);
+                log.info(
+                    "{} has {} tablets whose file sizes are not yet <= {}. Avg. offending file size: {}",
+                    tableName, offendingTabletSizes.size(), splitThreshold,
+                    String.format("%.0f", averageFileSize));
+              }
+              MILLISECONDS.sleep(sleepMillis);
+            }
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        });
+
+        long timeTaken = System.nanoTime() - beforeThresholdUpdate;
+
+        log.info(
+            "Time taken for all tables to reach expected total file size ({}): {} seconds ({}ms)",
+            splitThresholdStr, NANOSECONDS.toSeconds(timeTaken), NANOSECONDS.toMillis(timeTaken));
+      }
+
+      log.info("Deleting tables");
+      tableNames.stream().parallel().forEach(tableName -> {
+        try {
+          client.tableOperations().delete(tableName);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      });
+      log.info("Deleting namespace");
+      client.namespaceOperations().delete(NAMESPACE);
+
+    }
+  }
+
+  /**
+   * @return a map of tablets to the sum of their file size
+   */
+  private static Map<Text,Long> getTabletFileSizes(AccumuloClient client, String tableName)
+      throws TableNotFoundException, AccumuloException, AccumuloSecurityException {
+    TableId tableId = TableId.of(client.tableOperations().tableIdMap().get(tableName));
+    try (Scanner scanner = client.createScanner(MetadataTable.NAME)) {
```

Comment:
Also had an observation/question here. It seems like using a BatchScanner with a list of only the tabets whos file size sum is not what we want would be more efficient here. I was wondering if there is a way to supply an inverse set of ranges to use. For this case we know the tablets that have already reached that condition so could we somehow construct a list of ranges around those tablets as opposed to a list of ranges on those tablets?

For this case I don't think it would be very practical because we are not particularly worried about efficiency here, plus the set of tablets is expected to change frequently. But was curious so I thought I would ask.